lgn-provers/src/provers/v1/groth16/mod.rs

```diff
@@ -25,22 +24,19 @@ pub fn create_prover(
     checksums: &HashMap<String, blake3::Hash>,
     pk_file: &str,
     vk_file: &str,
-) -> anyhow::Result<Groth16<impl Prover>> {
+) -> anyhow::Result<impl LgnProver> {
+    #[cfg(feature = "dummy-prover")]
+    let prover = {
+        info!("Creating dummy groth16 prover");
+        dummy_prover::DummyProver
+    };
+
+    #[cfg(not(feature = "dummy-prover"))]
     let prover = {
-        #[cfg(feature = "dummy-prover")]
-        let prover = {
-            info!("Creating dummy groth16 prover");
-            dummy_prover::DummyProver
-        };
-        #[cfg(not(feature = "dummy-prover"))]
-        let prover = {
-            info!("Creating groth16 prover");
-            euclid_prover::Groth16Prover::init(url, dir, circuit_file, pk_file, vk_file, checksums)?
-        };
-
-        debug!("Groth16 prover created");
-        prover
+        info!("Creating groth16 prover");
+        euclid_prover::Groth16Prover::init(url, dir, circuit_file, pk_file, vk_file, checksums)?
     };
 
-    Ok(Groth16::new(prover))
```

Comment:
Each prover had a wrapper struct that dispatched the messages to the inner prover. Now these are no longer needed and the prover is returned directly. This is one layer of indirection that was removed.